dsrag/vector_db.py

```diff
@@ -421,4 +423,100 @@ def to_dict(self):
         return {
             **super().to_dict(),
             'kb_id': self.kb_id,
-        }
\ No newline at end of file
+        }
+
+
+class MilvusDB(VectorDB):
+    def __init__(self, kb_id: str, storage_directory: str = '~/dsRAG', dimension: int = 768):
+        self.kb_id = kb_id
+        # Expand user path and ensure directory exists
+        self.storage_directory = os.path.expanduser(storage_directory)
+        self.vector_storage_path = os.path.join(self.storage_directory, 'vector_storage')
+        os.makedirs(self.vector_storage_path, exist_ok=True)  # Create directory if it doesn't exist
+
+        # Initialize Milvus client
+        self.client = MilvusClient(uri=f"{self.vector_storage_path}.db")
+        self._create_collection(collection_name=self.kb_id, dimension=dimension)
```

Comment:
The __init__ function gets called when loading an existing KnowledgeBase (and VectorDB) object, in addition to when creating a new one, so it needs to be able to handle both scenarios. Currently this code will just delete an existing collection if it exists, rather than loading it.